harvester/dmat/src/main/java/dk/dbc/dataio/harvester/dmat/HarvestOperation.java

```diff
@@ -62,13 +84,32 @@ public HarvestOperation(DMatHarvesterConfig config,
         this.dmatServiceConnector = dmatServiceConnector;
         this.jsonbContext = new JSONBContext();
         this.timezone = getTimezone();
+        this.rawRepoConnector = rawRepoConnector != null
+                ? rawRepoConnector
+                : createRawRepoConnector(config);
+    }
+
+    private RawRepoConnector createRawRepoConnector(DMatHarvesterConfig config) {
+        return new RawRepoConnector(config.getContent().getResource());
+    }
+
+    RecordServiceConnector createRecordServiceConnector() throws HarvesterException {
+        try {
+            final String recordServiceUrl = rawRepoConnector.getRecordServiceUrl();
+            LOGGER.info("Using record service URL: {}", recordServiceUrl);
+            return RecordServiceConnectorFactory.create(recordServiceUrl);
+        } catch (SQLException | QueueException | ConfigurationException e) {
+            throw new HarvesterException("Unable to obtain record service URL", e);
+        }
     }
 
     public int execute() throws HarvesterException {
         final StopWatch stopwatch = new StopWatch();
         final Map<Integer, Status> statusAfterExport = new HashMap<>();
         int recordsHarvested = 0;
 
+        RecordServiceConnector recordServiceConnector = createRecordServiceConnector();
```

Comment:
Behøver du jo sådan set kun initialisere ved start. Og ikke ved hver execute.

Reply:
Isoleret set korrekt, men så kan vi ikke injecte en recordService connector - og constructoren springer i luften fordi vi kalder createRecordServiceConnector() før vi har mulighed for at mocke den..   det kræver det færdige HarvestOperation objekt.